modules/yieldbotBidAdapter.js

```diff
@@ -0,0 +1,806 @@
+  initAdRequestParams: function(adRequestId, bidRequest) {
+    let commonSearchParams = {};
+    commonSearchParams[this.CONSTANTS.REQUEST_PARAMS.ADAPTER_VERSION] = this.CONSTANTS.VERSION;
+    commonSearchParams[this.CONSTANTS.REQUEST_PARAMS.USER_ID] = bidRequest.data.vi || this.CONSTANTS.VERSION + '-vi';
+    commonSearchParams[this.CONSTANTS.REQUEST_PARAMS.SESSION_ID] = bidRequest.data.si || this.CONSTANTS.VERSION + '-si';
+    commonSearchParams[this.CONSTANTS.REQUEST_PARAMS.PAGEVIEW_ID] = bidRequest.data.pvi || this.CONSTANTS.VERSION + '-pvi';
+    commonSearchParams[this.CONSTANTS.REQUEST_PARAMS.AD_REQUEST_ID] = adRequestId;
+    return commonSearchParams;
+  },
+
+  buildAdUrl: function(urlPrefix, publisherNumber, commonSearchParams, bid) {
+    const searchParams = Object.assign({}, commonSearchParams);
+    searchParams[this.CONSTANTS.REQUEST_PARAMS.BID_RESPONSE_TIME] = Date.now();
+    searchParams[this.CONSTANTS.REQUEST_PARAMS.AD_REQUEST_SLOT] = bid.slot + ':' + bid.size;
+    searchParams[this.CONSTANTS.REQUEST_PARAMS.INTERSECTION_OBSERVER_AVAILABLE] = this.intersectionObserverAvailable(window);
+
+    const queryString = buildQueryString(searchParams) || '';
+    const adUrl = urlPrefix +
+            publisherNumber +
+            this.CONSTANTS.REQUEST_API_PATH_CREATIVE +
+            '?' +
+            queryString;
+    return adUrl;
+  },
+
+  buildImpressionUrl: function(urlPrefix, publisherNumber, commonSearchParams) {
+    const searchParams = Object.assign({}, commonSearchParams);
+    const queryString = buildQueryString(searchParams) || '';
+    const impressionUrl = urlPrefix +
+            publisherNumber +
+            this.CONSTANTS.REQUEST_API_PATH_IMPRESSION +
+            '?' +
+            queryString;
+    return impressionUrl;
+  },
+
+  /**
+   * Object with Yieldbot ad markup representation and unique creative identifier.
+   * @typeDef {TagObject} TagObject
+   * @type {object}
+   * @property {string} creativeId bidder specific creative identifier for tracking at the source
+   * @property {string} ad ad creative markup
+   * @memberof module:YieldbotBidAdapter
+   */
+  /**
+   * Builds the ad creative markup.
+   * @param {string} urlPrefix base url for Yieldbot requests
+   * @param {module:YieldbotBidAdapter.YieldbotBid} bid Bidder slot bid object
+   * @returns {module:YieldbotBidAdapter.TagObject}
+   * @memberof module:YieldbotBidAdapter
+   */
+  buildAdCreativeTag: function(urlPrefix, bid, bidRequest) {
+    const ybotAdRequestId = this.newId();
+    const commonSearchParams = this.initAdRequestParams(ybotAdRequestId, bidRequest);
+    const publisherNumber = bidRequest && bidRequest.yieldbotSlotParams ? bidRequest.yieldbotSlotParams.psn || '' : '';
+    const adUrl = this.buildAdUrl(urlPrefix, publisherNumber, commonSearchParams, bid);
+    const impressionUrl = this.buildImpressionUrl(urlPrefix, publisherNumber, commonSearchParams);
+
+    let htmlMarkup = `<div id="ybot-${ybotAdRequestId}"></div>
```

Comment:
@solenoid as mentioned I left the html markup string "pretty" for readability.

@mkendall07 regarding adapter module size we could minify the creative rendering script markup as I see others do. However, at least for the PR, I think it's worth making it readable.

Additionally will make a comment wrt module size here:
 - [L78](https://github.com/prebid/Prebid.js/pull/2135/files#diff-880a65bced0c70c4766ea1c6ef18471dR78)

Reply:
@elljoh makes good sense and I'll see about how to mitigate size concerns while leaving legibility good for review.

Reply:
@solenoid stripping whitespace here will save approx. 1K  so I will do that before any lgtm. Just wanted to make sure at least the snippet was readable for initial reviews.

Reply:
@elljoh does the uglify.js process not strip these out in the final build?

Reply:
@mkendall07 not it does not b/c the snippet is a string literal.

Reply:
@elljoh and @mkendall07 I like the simple whitespace stripping approach and also doing more involved minifying inside the string literal only if strictly needed since it would reduce readability a fair amount more.